doc/src/learn/cryptography/groth16.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Zero-knowledge proof verification (Groth16)
+---
+
+## Overview
+
+A zero-knowledge proof allows a prover to prove that a statement is true without revealing any information about the inputs, for example a prover can prove that they know the solution to a sudoku puzzle without revealing the solution.
+
+Zk-SNARKs (Zero-Knowledge Succinct Non-Interactive Argument of Knowledge) are a family of zero-knowledge proofs which are non-interactive, have succinct proof size and efficient verification time. 
+An important and widely used variant of them is pairing-based zk-SNARKs, of which the [Groth16](https://eprint.iacr.org/2016/260.pdf) proof system is one of the most efficient and widely used.
+
+The Move API in Sui allows users to verify any statement that can be expressed in a NP-complete language can be verified efficiently using Groth16 zk-SNARKs over either the Bn254 or BLS12-381 elliptic curve constructions.
+
+There are high-level languages for expressing these statements, such as [Circom](https://docs.circom.io) which will be used in the example below.
+
+Note that Groth16 requires a trusted setup for each circuit to generate the verification key, but we will not discuss how to do that here.
+
+## Usage
+
+The following example demonstrates how to create a Groth16 proof from a statement written in Circom and then verify it using the Sui Move API. 
```

Comment:
Add a clear note that our verifier currently supports up to 8 public inputs in the zk circuit. This is important info.